ui/mirage/config.js

```diff
@@ -312,6 +313,65 @@ export default function() {
   this.get('/client/allocation/:id/stats', clientAllocationStatsHandler);
   this.get('/client/fs/logs/:allocation_id', clientAllocationLog);
 
+  this.get('/client/fs/ls/:allocation_id', (schema, { queryParams }) => {
+    if (queryParams.path.endsWith('empty-directory')) {
+      return [];
+    } else if (queryParams.path.endsWith('directory')) {
+      return [
+        {
+          Name: 'another',
+          IsDir: true,
+          ModTime: moment().format(),
+        },
+      ];
+    } else if (queryParams.path.endsWith('another')) {
+      return [
+        {
+          Name: 'something.txt',
+          IsDir: false,
+          ModTime: moment().format(),
+        },
+      ];
+    } else {
+      return [
+        {
+          Name: '🤩.txt',
+          IsDir: false,
+          Size: 1919,
+          ModTime: moment()
+            .subtract(2, 'day')
+            .format(),
+        },
+        {
+          Name: '🙌🏿.txt',
+          IsDir: false,
+          ModTime: moment()
+            .subtract(2, 'minute')
+            .format(),
+        },
+        {
+          Name: 'directory',
+          IsDir: true,
+          Size: 3682561,
+          ModTime: moment()
+            .subtract(1, 'year')
+            .format(),
+        },
+        {
+          Name: 'empty-directory',
+          IsDir: true,
+          ModTime: moment().format(),
+        },
+      ];
+    }
+  });
```

Comment:
I don't think we need a full blown factory for this since it's such a specific API, but I'm wondering if there's a better place to stuff this fixture. Mirage has a fixtures concept in addition to factories, but I don't know how well the two play together.

Reply:
I bypassed the Mirage ORM entirely with this, which is why it doesn’t use factories or fixtures. I did have visions of some kind of file/directory data structure situation, but it seemed overwrought. I don’t love having this giant handler in the Mirage configuration but I thought it was better to have it present for demonstration purposes.